xls/examples/sobel_filter.x

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021 The XLS Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Floating point sobel filter implementation (fixed-point is TODO).  
+// A sobel filter, used in image and video processing, applies stencils
+// on an image to sharpen edges.
+
+import std
+import float32
+import xls.modules.fpadd_2x32
+import xls.modules.fpmul_2x32
+import third_party.xls_go_math.fpsqrt_32
+
+type F32 = float32::F32;
+
+// Sobel stencils.
+const X_STENCIL = s32[3][3]:[s32[3]:[-1,  0,  1],
+                             s32[3]:[-2,  0,  2],
+                             s32[3]:[-1,  0,  1]]; 
+const Y_STENCIL = s32[3][3]:[s32[3]:[-1, -2, -1],
+                             s32[3]:[ 0,  0,  0],
+                             s32[3]:[ 1,  2,  1]]; 
+
+// Do we have a way to reshape multi-dimensional arrays or to apply map to
+// multidemnsional arrays? 
```

Comment:
Typo ("multidimensional")

Reply:
Also this sort of Q should probably be a formal TODO or be addressed before submit. @cdleary : I don't believe that we do, but I want your brain.

Reply:
Done.